File: C++/213. House Robber II.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        if(n==1){return nums[0];}
        else{
            vector<int> vec1(n,0);
            vector<int> vec2(n,0);
            vec1[1]=nums[0];
            vec2[1]=nums[1];
            for(int i = 2; i<n; i++){
                vec1[i] = max(vec1[i-1], vec1[i-2] + nums[i-1]);
                vec2[i] = max(vec2[i-1], vec2[i-2] + nums[i]);
            }
            return max(vec1[n-1], vec2[n-1]);
        }
    }
};

File: C#/213. House Robber II.cs
public class Solution {
    public int Rob(int[] nums) {
        int n = nums.Length;
        if(n==1){return nums[0];}
        else{
            int[] vec1 = new int[n];
            int[] vec2 = new int[n];
            vec1[1]=nums[0];
            vec2[1]=nums[1];
            for(int i = 2; i<n; i++){
                vec1[i] = Math.Max(vec1[i-1], vec1[i-2] + nums[i-1]);
                vec2[i] = Math.Max(vec2[i-1], vec2[i-2] + nums[i]);
            }
            return Math.Max(vec1[n-1], vec2[n-1]);
        }
    }
}
